pyaedt/edb_core/EDB_Data.py

```diff
@@ -3241,6 +3482,111 @@ def top_bottom_association(self):
+
+        with open(file_path, "r") as f:
+            for line in f:
+                if "subckt" in line.lower():
+                    pinNames = [i.strip() for i in re.split(" |\t", line) if i]
+                    pinNames.remove(pinNames[0])
+                    pinNames.remove(pinNames[0])
+                    break
+        if len(pinNames) == self.numpins:
+            model = self._edb.Cell.Hierarchy.SPICEModel()
+            model.SetModelPath(file_path)
+            model.SetModelName(name)
+            terminal = 1
+            for pn in pinNames:
+                model.AddTerminalPinPair(pn, str(terminal))
+                terminal += 1
+        else:  # pragma: no cover
+            logging.error("Wrong number of Pins")
+            return False
+        return self._set_model(model)
+
+    @pyaedt_function_handler
+    def assign_s_param_model(self, file_path, name=None, reference_net=None):
+        """Assign S-parameter to this component.
+
+        Parameters
+        ----------
+        file_path : str
+            File path of the S-parameter model.
+        name : str, optional
+            Name of the S-parameter model.
+        Returns
+        -------
+
+        """
+        if not name:
+            name = get_filename_without_extension(file_path)
+
+        edbComponentDef = self.edbcomponent.GetComponentDef()
+        nPortModel = self._edb.Definition.NPortComponentModel.FindByName(edbComponentDef, name)
+        if nPortModel.IsNull():
+            nPortModel = self._edb.Definition.NPortComponentModel.Create(name)
+            nPortModel.SetReferenceFile(file_path)
+            edbComponentDef.AddComponentModel(nPortModel)
+
+        model = self._edb.Cell.Hierarchy.SParameterModel()
+        model.SetComponentModelName(name)
+        if reference_net:
+            model.SetReferenceNet(reference_net)
+        return self._set_model(model)
+
+    @pyaedt_function_handler
+    def assign_rlc_model(self, res, ind, cap, is_parallel=False):
+        """Assign RLC to this component.
+
+        Parameters
+        ----------
+        res : int, float
+            Resistance.
+        ind : int, float
+            Inductance.
+        cap : int, float
+            Capacitance.
+        is_parallel : bool
+            Whether it is parallel or series RLC component.
+        """
+        res, ind, cap = self._get_edb_value(res), self._get_edb_value(ind), self._get_edb_value(cap)
+        model = self._edb.Cell.Hierarchy.PinPairModel()
+
+        pin_names = list(self.pins.keys())
+        for idx, i in enumerate(np.arange(len(pin_names) // 2)):
+            pin_pair = self._edb.Utility.PinPair(pin_names[idx], pin_names[idx + 1])
+            rlc = self._edb.Utility.Rlc(res, True, ind, True, cap, True, is_parallel)
+            model.SetPinPairRlc(pin_pair, rlc)
+        self._set_model(model)
+
 
 class EdbBuilder(object):
     """Data Class to Overcome EdbLib in Linux."""
```

Comment:
Format for a class description must begin with a verb ending in an "s". Perhaps change to: "Provides a data class to overcome ``EdbLib`` class limitations in Linux."